src/Parallel/Main.hpp

```diff
@@ -799,6 +808,60 @@ void Main<Metavariables>::add_exception_message(std::string exception_message) {
       });
 }
 
+template <typename Metavariables>
+void Main<Metavariables>::did_all_elements_terminate(
+    const bool all_elements_terminated) {
+  if (not all_elements_terminated) {
+    if (UNLIKELY(current_termination_check_index_ == 0)) {
```

Comment:
If you make the above change about `check_if_next_component_terminated_properly`, you can get rid of this check here.